src/ast/LlvmVisitor.java

```diff
@@ -109,67 +163,232 @@ public void visit(MainClass mainClass) {
+        /**
+         * Case 1: assignee is a local variable in the method
+         */
+        if (enclosingMethodST.contains(assigneeName, false)) {
+            assigneePtrReg = methodCurrRegIndex++;
+            LLVMProgram = LLVMProgram.concat(String.format(
+                    "\t%%_%d = load i32*, i32** %%%s", assigneePtrReg, assigneeName
+            ));
+        }
+        /**
+         * Case 2: assignee is a field of %this
+         */
+        else if (classInstanceHasField(classInstanceShape, assigneeName)) {
+            assigneePtrReg = methodCurrRegIndex;
+            int offset = calcFieldOffset(classInstanceShape, assigneeName);
+            LLVMProgram = LLVMProgram.concat(String.format(
+                    "\t%%_%d = getelementptr i8, i8* %this, i32 %d\n", assigneePtrReg, offset
+            ));
+            methodCurrRegIndex++;
+            int assigneePtrRegPostCast = methodCurrRegIndex;
+            LLVMProgram = LLVMProgram.concat(String.format(
+                    "\t%%_%d = bitcast i8* %%_%d to %s*\n", assigneePtrRegPostCast, assigneePtrReg, assigneeLLType
+            ));
+            methodCurrRegIndex++;
+            assigneePtrReg = assigneePtrRegPostCast;
+        } else return;
+        /**
+         * Check that index is legal, throw exception / assign
+         */
+        assignArrayStatement.index().accept(this);
+        int indexReg = methodCurrRegIndex-1;
+        // Check that the index is greater than zero
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\t%%_%d = icmp slt i32 %d, 0\n", methodCurrRegIndex++, indexReg
+        ));
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\tbr i1 %%_%d, label %%arr_alloc%d, label %arr_alloc%d\n", methodCurrRegIndex-1, methodCurrLabelIndex, methodCurrLabelIndex +1
+        ));
+        // Else throw out of bounds exception
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "arr_alloc%d:\n", methodCurrLabelIndex++
+        ));
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\tcall void @throw_oob()\n\tbr label %%arr_alloc%d\n", methodCurrLabelIndex
+        ));
+        // ok, continue. Load the size of the array (first integer of the array)
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "arr_alloc%d:\n", methodCurrLabelIndex++
+        ));
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\t%%_%d = getelementptr i32, i32* %%_%d, i32 0\n", methodCurrRegIndex++, assigneePtrReg
+        ));
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\t%%_%d = load i32, i32* %%_%d\n", methodCurrRegIndex, methodCurrRegIndex-1
+        ));
+        methodCurrRegIndex++;
+        // Check that the index is less than the size of the array
+        LLVMProgram = LLVMProgram.concat(String.format(
+                "\t%%_%d = icmp sle i32 %%_%d, %%_%d", methodCurrRegIndex, methodCurrRegIndex-1, indexReg
```

Comment:
we need to check that index < size.
<= is not good because size itself is out of bounds